src/emcpy/utils/dateutils.py

```diff
@@ -0,0 +1,226 @@
+# coding: utf-8 -*-
+
+"""
+utilities for working with dates using datetime module (Python 2.3 or later)
+original author: Jeff Whitaker <jeffrey.s.whitaker@noaa.gov>
+"""
+
+__all__ = [
+    "dateto_hrs_since_day1CE",
+    "hrs_since_day1CE_todate",
+    "dateshift",
+    "splitdate",
+    "makedate",
+    "hrstodate",
+    "datetohrs",
+    "daterange",
+    "dayofyear",
+    "getyrmon",
+    "daysinmonth",
+]
+
+import datetime, calendar
+
+
+hrsgregstart = 13865688  # hrs from 00010101 to 15821015 in Julian calendar.
+# times in many datasets use mixed Gregorian/Julian calendar, datetime
+# module uses a proleptic Gregorian calendar. So, I use datetime to compute
+# hours since start of Greg. calendar (15821015) and add this constant to
+# get hours since 1-Jan-0001 in the mixed Gregorian/Julian calendar.
+gregstart = datetime.datetime(1582, 10, 15)  # datetime.datetime instance
+day1 = datetime.datetime(1, 1, 1)  # datetime.datetime instance
+
+
+def dateto_hrs_since_day1CE(curdate, mixedcal=True):
+    """
+    Given datetime.datetime instance, compute hours since 1-Jan-0001
+    Args:
+        curdate: (datetime.datetime instance) date to check.
+        mixedcal: (bool) is mixed Gregorian/Julian calendar used?
+
+    Returns:
+        The number of hours since 1-Jan-0001
+    """
+    if mixedcal:
+        if curdate < gregstart:
+            msg = "date must be after start of gregorian calendar (15821015)!"
+            raise ValueError(msg)
+        difftime = curdate - gregstart
+        hrsdiff = 24 * difftime.days + difftime.seconds / 3600
+        return hrsdiff + hrsgregstart
+    else:
+        difftime = curdate - day1
+        return 24.0 * (difftime.days + 1) + difftime.seconds / 3600.0
+
+
+def hrs_since_day1CE_todate(hrs, mixedcal=True):
+    """
+    return datetime.datetime instance given hours since 1-Jan-0001
+
+    Args:
+        hrs: (float) number of hours since 1-Jan-0001 to compute a date.
+        mixedcal: (bool) is mixed Gregorian/Julian calendar used?
+    Returns:
+        curdate: (datetime.datetime instance) date given hours since 1-Jan-0001
+    """
+    if hrs < 0.0:
+        msg = "hrs must be positive!"
+        raise ValueError(msg)
+    delta = datetime.timedelta(hours=1)
+    if mixedcal:
+        hrs_sincegreg = hrs - hrsgregstart
+        curdate = gregstart + hrs_sincegreg * delta
+    else:
+        curdate = hrs * delta
+    return curdate
+
+
+def dateshift(analdate, fcsthr):
+    """
+    compute verification date given analysis date string (yyyymmddhh) and fcst hour.
+    Args:
+        analdate: (date str) analysis date
+        fcsthr: (int) forecast hour
+    Returns:
+        verifdate: (date str) verification date given analysis date string (yyyymmddhh) and fcst hour.
+    """
+    yyyy, mm, dd, hh = splitdate(analdate)
+    analdate = datetime.datetime(yyyy, mm, dd, hh)
+    verifdate = analdate + fcsthr * datetime.timedelta(hours=1)
```

Comment:
why not `hours=fcsthr`?

Reply:
I updated the calculation to do
verifdate = analdate +  datetime.timedelta(hours=fcsthr)
either way should be correct though.